### ISSUE TITLE
 val loss log not show using IterBasedRunner

### ISSUE THREAD

**Issue by fangwudi** [open-mmlab/mmdetection#5382]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
config like below:
dataset_type = 'MyCustomDataset'
workflow = [('train', 1), ('val', 1)]
runner = dict(type='IterBasedRunner' , max_iters=10)

sh dist_train.sh and result (val loss not print out):
![image](https://user-images.githubusercontent.com/20792612/122321654-c984e900-cf56-11eb-8fea-b75325826204.png)

By the way, when I use EpochBasedRunner, it works fine

**Reproduction**

1. What command or script did you run?

```none
A placeholder for the command.
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use?

**Environment**

TorchVision: 0.6.1+cu101
OpenCV: 4.5.1
MMCV: 1.3.6
MMCV Compiler: GCC 7.5
MMCV CUDA Compiler: 10.1
MMDetection: 2.13.0+f34a9a1

**Error traceback**
If applicable, paste the error trackback here.

```none
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by alpereninci** [comment]
I have the same problem.

**Comment by alpereninci** [comment]
Since "after_val_iter" is not implemented in mmcv/runner/hooks/logger/base.py, call_hook("after_val_iter") is passed. To solve the problem, I copied def after_train_iter(self,runner): function under the name def after_val_iter(self,runner). For now, it seems okey. Please inform me, if you see any problem of this solution. 

![image](https://user-images.githubusercontent.com/44204665/123555423-50736600-d78e-11eb-8b31-93aad54f7daf.png)

**Comment by hhaAndroid** [comment]
Hi @fangwudi This is a bug, we will fix it as soon as possible.

**Comment by fangwudi** [comment]
> Hi @fangwudi This is a bug, we will fix it as soon as possible.

thanks.
